app/src/main/java/net/squanchy/contest/ContestActivity.java

```diff
@@ -0,0 +1,66 @@
+package net.squanchy.contest;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.util.Locale;
+
+import net.squanchy.R;
+import net.squanchy.fonts.TypefaceStyleableActivity;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+
+public class ContestActivity extends TypefaceStyleableActivity {
+
+    private static final String EXTRA_CHECKPOINT_ID = ContestActivity.class.getCanonicalName() + ".checkpoint_id";
+
+    private ContestService contestService;
+    private TextView contestResults;
+
+    public static Intent createIntent(Context context, String checkpoint) {
```

Comment:
❌  if the `checkpoint` (which you call `achievementId` elsewhere, please uniform names) is optional you should either have two overloads of this method, one that doesn't and one that does take it as an argument, or just one that takes `Optional<String> achievementId`